### DISCUSSION TITLE
Retain context when jumping from exemplar in dashboard to trace

### DISCUSSION THREAD

**Discussion by 09jvilla** [grafana/grafana#65310] [Dashboards, Panels & Visualizations]
One user has brought up that they lose context when jumping from a dashboard panel with exemplars into the trace pointed to by that exemplar. 

The way the interaction works today is that the user hovers over the dot representing the exemplar, which opens a modal. From there, they click the "query in tempo" button, which takes them directly to an explore view that displays the trace. This means that they lose the context of the dashboard and metric panel they were previously looking at. They can only get that context back by hitting the back button in their browser. Alternative they can open a new tab from "query in tempo" and then flip between tabs, but they also feel this causes them to lose context. 

Their suggestion to help maintain context is to change the interaction so that "query in tempo" splits the screen between the dashboard on one side and the trace view on the other. This is similar to the interaction today in Explore. However, its not clear to me that this proposed interaction is a pattern that is used anywhere else in dashboards, which means it might feel jarring for others. One alternative to consider could also be having the "query in tempo" button in a dashboard open a split screen view in the Explore, with the metric on one side and the trace on the other (instead of going to an Explore view with just the trace). 

(Below) Current exemplar dashboard interaction that causes the user to lose context:

https://user-images.githubusercontent.com/9610816/227397584-32f3902c-403d-4912-9620-9f714d6938cc.mp4

(Below) Exemplar explore interaction which allows the user to retain context via split screen interaction

https://user-images.githubusercontent.com/9610816/227397640-69c4e636-7ae8-49e6-8f26-d416a18f81c3.mp4

**Comment by thanos-karachalios** [comment]
This is a use case that appears to mainly focus on improving the user experience within a specific context of use.

Although the idea of dividing the screen between the dashboard panel and trace may seem interesting at first glance, it may not align with the purpose of dashboards and may add unnecessary complexity without significant benefits, particularly in terms of interaction. Additionally, there hasn't been widespread demand for this kind of functionality from users.

On the other hand, Explore provides a better-suited environment for addressing these types of situations. As mentioned by @09jvilla above, the proposed solution follows these steps to mitigate the loss of initial context:

1. The user hovers over the dot that represents the example, which opens a modal.
2.From there, they click the "query in tempo" button.
3.This takes them to a split-screen view in Explore, where the metric is on one side and the trace is on the other, instead of simply directing them to an Explore view with only the trace.

cc @Elfo404

**Reply by Elfo404** [reply]
+1 on what @thanos-karachalios said.

i think this use case could fit quite nicely with correlations. In the meantime a workaround could be to open the query in explore (by clicking on explore in the panel menu) and then follow the link from there.

**Reply by Elfo404** [reply]
cc. @ifrost

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82762. Feel free to continue the discussion around this there. Thank you!
